src/ex2/CompileVisitor.java

```diff
@@ -196,19 +197,135 @@ public void visit(WhileStatement whileStatement) {
     // ARRAYS
     @Override
     public void visit(AssignArrayStatement assignArrayStatement) {
+        var arrayName = assignArrayStatement.lv();
+        var arrayRegister = lastRegisterNumber++;
+        addLine(String.format("%%_%s = load i32*, i32** %%%s", arrayRegister, arrayName));
+
+        assignArrayStatement.index().accept(this);
+        var indexRegister = lastRegisterNumber-1;
+        var indexComparisonToZeroRegister = lastRegisterNumber++;
+        addLine(String.format("%%_%s = icmp slt i32 %%_%s, 0", indexComparisonToZeroRegister, indexRegister));
+
+        var label1 = lastLabelNumber++;
+        var label2 = lastLabelNumber++;
+        addLine(String.format("br i1 %%_%s, label %%arr_check%s, label %%arr_check%s", indexComparisonToZeroRegister, label1, label2));
+        addLine(String.format("arr_check%s:", label1));
```

Comment:
function addLine also puts a tab at the start and it's be easier to read if labels do not have tabs

Reply:
Ok, will fix